### PR TITLE
Pair sign in

### PR DESCRIPTION
**Pull request by chrisjoywebb**

### CODE REVIEW

File: app/models/student.rb [hunk 1]
```diff
@@ -19,6 +19,14 @@ class Student < User
   belongs_to :primary_payment_method, class_name: 'PaymentMethod'
   has_many :signatures
 
+  def pair_on_day(day)
+    Student.find_by_id(attendance_record_on_day(day).try(:pair_id))
```
Comment (chrisjoywebb) [inline app/models/student.rb]: show michael problem

File: app/models/student.rb [hunk 2]
```diff
@@ -19,6 +19,14 @@ class Student < User
...
+  end
+
+  def attendance_record_on_day(day)
+    attendance_records.where(date: day).first
```
Comment (chrisjoywebb) [inline app/models/student.rb]: find_by(date: day)

File: app/controllers/attendance_records_controller.rb [hunk 3]
```diff
@@ -29,15 +38,15 @@ def update
...
+    AttendanceRecord.destroy(params[:ids])
+    flash[:alert] = params[:ids].count == 1 ? "Attendance record has been deleted." : "Attendance records have been deleted."
+    redirect_to attendance_path
   end
 
 private
 
   def attendance_record_params
-    params.require(:attendance_record).permit(:signing_out, :student_id)
+    params.require(:attendance_record).permit(:signing_out, :student_id, :pair_ids)
```
Comment (chrisjoywebb) [inline app/controllers/attendance_records_controller.rb]: take out student_id

File: app/models/attendance_record.rb [hunk 4]
```diff
@@ -3,6 +3,7 @@ class AttendanceRecord < ActiveRecord::Base
   scope :today, -> { where(date: Time.zone.now.to_date) }
   validates :student_id, presence: true, uniqueness: { scope: :date }
   validates :date, presence: true
+  validates :pair_id, uniqueness: { scope: [:student_id, :date] }
```
Comment (chrisjoywebb) [inline app/models/attendance_record.rb]: needs a test

File: app/controllers/attendance_records_controller.rb (hunk 3 above)
Comment (chrisjoywebb) [inline app/controllers/attendance_records_controller.rb]: done

File: app/models/student.rb (hunk 1 above)
Comment (chrisjoywebb) [inline app/models/student.rb]: done

File: app/models/student.rb (hunk 2 above)
Comment (chrisjoywebb) [inline app/models/student.rb]: done

File: app/models/attendance_record.rb (hunk 4 above)
Comment (chrisjoywebb) [inline app/models/attendance_record.rb]: done

File: app/controllers/pair_attendance_records_controller.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+class PairAttendanceRecordsController < ApplicationController
+
+  def create
+    @attendance_records = params[:pair_ids].map do |pair_id|
+      AttendanceRecord.new({
+        student_id: pair_id,
+        pair_id: params[:pair_ids].reverse.slice(params[:pair_ids].index(pair_id))
+      })
+    end
+    if @attendance_records.each { |record| record.save } && @attendance_records.first.id != nil
```
Comment (michaelrkn) [inline app/controllers/pair_attendance_records_controller.rb]: change to `if @attendance_records.all? { |record| record.save }

File: config/initializers/inflections.rb [hunk 6]
```diff
@@ -14,3 +14,7 @@
 # ActiveSupport::Inflector.inflections(:en) do |inflect|
 #   inflect.acronym 'RESTful'
 # end
+
+ActiveSupport::Inflector.inflections do |inflect|
+  inflect.irregular "has", "have"
```
Comment (michaelrkn) [inline config/initializers/inflections.rb]: nice!

File: app/controllers/pair_attendance_records_controller.rb (hunk 5 above)
Comment (michaelrkn) [inline app/controllers/pair_attendance_records_controller.rb]: After this, go ahead and merge!

Comment (chrisjoywebb) [inline app/controllers/pair_attendance_records_controller.rb]: done

**Pull request merged**